### PR TITLE
Query Module: Executor 

### PR DESCRIPTION
**Pull request by lujiajing1126**
This PR adds Executor for logical plan.

### The API of `index.Repo`

```go
type Repo interface {
        // 1) which Metadata should be used here, since metadata of IndexRule cannot be used to uniquely determine the traceSeries
        // So for now, I use metadata of `TraceSeries`
        // 2) Condition
	Search(index common.Metadata, startTime, endTime uint64, conditions []Condition) ([]common.ChunkID, error)
}
```

### Schema

The entity to be built should fully comply with the schema defined, in particular for the sparse fields.

It is not constraint currently.

### Test Dataset

We need to add a dataset for testing and performance benchmark in the near future. 

So far, I only use mock data.

### CODE REVIEW

**Comment by hanahmily** [comment]
Cool! I will review the details soon.

> We need to add a dataset for testing and performance benchmark in the near future.

After integrating with skywalking OAP, we can record the online/real dataset which is the best choice for the testing purpose.

File: api/common/id.go [hunk 1]
```diff
@@ -18,3 +18,40 @@
...
 
 type ChunkID uint64
+
+type ChunkIDs []ChunkID
+
+// HashIntersect returns an intersection of two ChunkID arrays
+// without any assumptions on the order. It uses a HashMap to mark
+// the existence of a item.
+func (c ChunkIDs) HashIntersect(other ChunkIDs) ChunkIDs {
+	if c == nil && other == nil {
```
Comment (hanahmily) [inline api/common/id.go:28]: The receiver `c` should not be nil from what I know. Can we remove this precondition?

File: pkg/logical/expr_literal.go [hunk 2]
```diff
@@ -52,12 +57,20 @@ func (i *int64Literal) String() string {
...
 
-var _ Expr = (*int64ArrLiteral)(nil)
+var _ LiteralExpr = (*int64ArrLiteral)(nil)
 
 type int64ArrLiteral struct {
 	arr []int64
 }
 
+func (i *int64ArrLiteral) Bytes() [][]byte {
+	var b [][]byte
```
Comment (hanahmily) [inline pkg/logical/expr_literal.go:67]: how about `make` a slice with the capacity as `i.arr`

**Review by hanahmily — COMMENTED** [review]
How about move `pkg/logical` and `pkg/execution` to `pkg/query/logical` and `pkg/query/execution` since they should be classified to `query` module.

File: api/common/id.go (hunk 1 above)
Comment (lujiajing1126) [reply api/common/id.go]: Fixed

File: pkg/logical/expr_literal.go (hunk 2 above)
Comment (lujiajing1126) [reply pkg/logical/expr_literal.go]: I've created a slice with capacity, for both `int64ArrLiteral` and `strArrLiteral`.

Later, we may add a general byte buffer module to optimize memory allocation.

**Comment by lujiajing1126** [comment]
> How about move `pkg/logical` and `pkg/execution` to `pkg/query/logical` and `pkg/query/execution` since they should be classified to `query` module.

Moved to `pkg/query/{logical,executor}`

**Pull request merged**